src/main.rs

```diff
@@ -348,6 +352,36 @@ fn find_ram_in_linker_script(linker_script: &str) -> Option<MemoryEntry> {
     None
 }
 
+fn arithmetic_op(line: &str) -> u64 {
+    let segments: Vec<&str> = line.split('+').map(|s| s.trim().trim_end()).collect();
+    // if some numbers are hex we convert here
+
+    let mut total_length = 0;
+    for segment in segments {
+        let boundary_pos = segment
+            .find(|c| c == 'K' || c == 'M')
+            .unwrap_or(segment.len());
+        const HEX: &str = "0x";
+        let length: u64 = if segment.starts_with(HEX) {
+            tryc!(u64::from_str_radix(&segment[HEX.len()..boundary_pos], 16))
+        } else {
+            tryc!(segment[..boundary_pos].parse())
+        };
+
+        let raw = &segment[boundary_pos..];
+        let mut chars = raw.chars();
+        let unit = chars.next();
+        if unit == Some('K') {
+            total_length += length * 1024;
+        } else if unit == Some('M') {
+            total_length += length * 1024 * 1024;
+        } else if unit == None {
```

Comment:
This can result in the resulting length being zero if there are other chars in the segment after the number. For example, with `LENGTH = 10K + 1 }` the last segment will be `1 }`, and thus `chars.next()` will return `Some(' ')` instead of `None`.